File: 二叉树/bfs 层数最深叶子节点的和.go
package 二叉树

func deepestLeavesSum0(root *TreeNode) int {
	var res int
	queue := []*TreeNode{root}
	for len(queue) != 0 {
		n := len(queue)
		curLayerSum := 0
		for i := 0; i < n; i++ {
			cur := queue[0]
			queue = queue[1:]
			curLayerSum += cur.Val
			if cur.Left != nil {
				queue = append(queue, cur.Left)
			}
			if cur.Right != nil {
				queue = append(queue, cur.Right)
			}
		}
		res = curLayerSum
	}
	return res
}

File: 二叉树/bfs层数最深叶子节点的和.java
package 二叉树;

import java.util.ArrayDeque;
import java.util.Queue;

public class bfs层数最深叶子节点的和 {
    int res;

    public int deepestLeavesSum(TreeNode root) {
        Queue<TreeNode> queue = new ArrayDeque<>(); //java内置的队列
        queue.offer(root);
        while (!queue.isEmpty()) {
            int n = queue.size();
            int curLayerSum = 0;
            for (int i = 0; i < n; i++) {
                TreeNode node = queue.poll();
                curLayerSum += node.val;
                if (node.left != null) {
                    queue.offer(node.left);
                }
                if (node.right != null) {
                    queue.offer(node.right);
                }
            }
            res = curLayerSum;
        }
        return res;
    }
}

